pkg/kubelet/cm/topologymanager/scope.go

```diff
@@ -162,3 +165,20 @@ func unexpectedAdmissionError(err error) lifecycle.PodAdmitResult {
 func admitPod() lifecycle.PodAdmitResult {
 	return lifecycle.PodAdmitResult{Admit: true}
 }
+
+func smtAlignmentError(err error) lifecycle.PodAdmitResult {
+	return lifecycle.PodAdmitResult{
+		Message: err.Error(),
+		Reason:  "SMTAlignmentError",
+		Admit:   false,
+	}
+}
+
+type SMTAlignmentError struct {
+	RequestedCPUs int
+	CpusPerCore   int
+}
+
+func (e SMTAlignmentError) Error() string {
+	return fmt.Sprintf("Number of CPUs requested should be a multiple of number of CPUs on a core = %d on this system. Requested CPU count = %d", e.CpusPerCore, e.RequestedCPUs)
+}
```

Comment:
It feels strange to me that this error type is defined here and not in the `CPUManager`.

Reply:
It was done because it was the simplest way to avoid circular imports, but yes, it should be inside the `CPUManager` somehow. We'll think about a better approach.